include/alpaka/dev/DevUniformCudaHipRt.hpp

```diff
@@ -43,7 +45,20 @@ namespace alpaka
     {
         template<typename TApi, bool TBlocking>
         class QueueUniformCudaHipRt;
-    }
+
+        inline auto getThreadPool() -> core::ThreadPool&
+        {
+#    if BOOST_COMP_CLANG
+#        pragma clang diagnostic push
+#        pragma clang diagnostic ignored "-Wexit-time-destructors"
+#    endif
+            static core::ThreadPool pool;
+#    if BOOST_COMP_CLANG
+#        pragma clang diagnostic pop
+#    endif
+            return pool;
+        }
+    } // namespace uniform_cuda_hip::detail
```

Comment:
After re-reading the changes I've been thinking... there is no particular reason why the thread pool should be bound to the CUDA/HIP device type, is it ?

Why not move this `getThreadPool()` as a `static` method of the `ThreadPool` class itself ?

And then use it directly from `enqueue()`, instead of making `enqueue()` call `sendTaskToThePool()` and `sendTaskToThePool()` use the thread pool ?

Reply:
This would introduce for the first time in alpaka a state which is not controlled by the user. 
If possible we should avoid such global state.

Reply:
I'm not sure I understand the difference ?

The current implementation uses a `static` data member in the CUDA or HIP device class.
My suggestion would use a `static` data member in the thread pool class.

And in both cases the pool is created only the first time it is needed.

Reply:
I was not aware that the current implementation is using a static variable to create the thread pool. 
IMO we should avoid using static states and applied for a change https://github.com/alpaka-group/alpaka/pull/1707/files#r867801735 but we should discuss it tomorrow in the developer meeting.